RNAndroidPlayground/app/src/main/java/com/kiwi/rnandroidplayground/MainActivity.kt

```diff
@@ -3,16 +3,27 @@ package com.kiwi.rnandroidplayground
 import android.content.Intent
 import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler
 import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
-
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_main)
 
     button_hotels.setOnClickListener {
       startActivity(Intent(this, HotelsActivity.getViewModelClass()))
     }
+
+    button_hotels_fragment.setOnClickListener {
+      startActivity(Intent(this, HotelActivityFragment.getViewModelClass()))
+    }
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    // TODO check where to use this
+    (application as PlaygroundApplication).reactNativeHost.reactInstanceManager.onHostDestroy(this)
```

Comment:
In production application it would be probably the best to initialize the RN instance in main's activity onCreate()/onDestroy() since Application's onDestroy is tend to be used only for debugging and is not called in production app. Possibly this can be called also in https://developer.android.com/reference/android/app/Application.html#onTrimMemory(int) when memory is low or critical and if user has currenlty no RN screen opened but this would require more logic to handle this state properly and to reinitialize the RN on demand.

Reply:
Mmm yes so I guess you are kinda "free" on where you want to init RN. I don't think it would hang Splash screen by the way given the fact that it's done in the background?

Reply:
Depends on if it is 100% async or some part of that call is synchronous. I haven't tried that, I am just currious.